hydroflow_datalog/src/grammar.rs

```diff
@@ -17,28 +17,36 @@ pub mod datalog {
     #[derive(Debug, Clone)]
     pub struct Rule {
         pub target: Atom,
+
         #[rust_sitter::leaf(text = ":-")]
         _from: (),
+
         #[rust_sitter::repeat(non_empty = true)]
         #[rust_sitter::delimited(
             #[rust_sitter::leaf(text = ",")]
             ()
         )]
         pub sources: Vec<Atom>,
+
+        pub filters: Option<RuleFilters>,
```

Comment:
This requires all filters to come after sources, but it'd be nice to instead make `Atom` an enum that has variants for both reading from relations or boolean conditions. That would also resolve the issue with needing different delimiters.